src/main/boilerplate/spray/json/ProductFormatsInstances.scala.template

```diff
@@ -34,6 +41,14 @@ trait ProductFormatsInstances { self: ProductFormats with StandardFormats =>
     def read(value: JsValue) = {
       [#val p1V = fromField[P1](value, fieldName1)#
       ]
+      if (!allowExtraKeys) {
+        val jsObject = value.asJsObject()
+        val keySet = jsObject.fields.keys.toSet
+        val keySetDiff = keySet.diff(knownFields)
+        if (!keySetDiff.isEmpty) {
+          throw new DeserializationException(s"${keySetDiff.head} is not a known key", null, keySetDiff.toList)
```

Comment:
Can we give more complete information when it fails? Missing set, given set and expected set of keys.